Assets/Scripts/MeleeAttack.cs

```diff
@@ -0,0 +1,28 @@
+﻿using UnityEngine;
+
+public class MeleeAttack : MonoBehaviour
```

Comment:
I don't think this should be a MonoBehaviour. Just have one MonoBehaviour that handles all attacks/spells and regular classes that describe the classes/spells.